### PR TITLE
cross-region: support to specify cluster name prefix

### PR DESCRIPTION
**Pull request by Thearas**
Signed-off-by: Thearas <thearas850@gmail.com>

### What problem does this PR solve?
We may run multiple cross-region tests concurrently in one namespace, but it's impossible for now because all tidb clusters created by cross-region has the same name.

### What is changed and how does it work?

Support `cluster-name` flag. It will be a prefix of every cluster(a.k.a. `dcLocation`) name.

See also:
The PR of adding `cluster-name` flag: https://github.com/pingcap/tipocket/pull/388.

### Check List

Tests

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: testcase/cross-region/tso.go [hunk 1]
```diff
@@ -70,17 +70,17 @@ func (c *crossRegionClient) testTSO(ctx context.Context) error {
...
 	}
 	log.Info("new allocators ready")
 	// after transfer allocator and leader, we can tolerate tso failed once for each dc.
 	c.requestTSOAfterTransfer(ctx, "global")
-	c.requestTSOAfterTransfer(ctx, "dc-1")
-	c.requestTSOAfterTransfer(ctx, "dc-2")
-	c.requestTSOAfterTransfer(ctx, "dc-3")
+	c.requestTSOAfterTransfer(ctx, DCLocations[0])
+	c.requestTSOAfterTransfer(ctx, DCLocations[1])
+	c.requestTSOAfterTransfer(ctx, DCLocations[2])
```
Comment (JmPotato) [inline testcase/cross-region/tso.go:83]: I think using a loop here to iter is better.

**Review by JmPotato — COMMENTED** [review]
The Rest LGTM.

File: testcase/cross-region/tso.go [hunk 2]
```diff
@@ -70,17 +70,17 @@ func (c *crossRegionClient) testTSO(ctx context.Context) error {
 		return err
 	}
 	log.Info("new leader ready")
-	for i := 1; i <= dcLocationNumber; i++ {
-		if err := c.transferPDAllocator(fmt.Sprintf("dc-%d", i)); err != nil {
+	for i := 0; i < dcLocationNumber; i++ {
```
Comment (zyguan) [inline testcase/cross-region/tso.go:73]: `len(DCLocations)` may be better.

File: testcase/cross-region/tso.go (hunk 1 above)
Comment (Thearas) [reply testcase/cross-region/tso.go:83]: Ack.

File: testcase/cross-region/cross-region.go [hunk 3]
```diff
@@ -34,6 +34,8 @@ add placement policy
 	role=leader
 	replicas=1`
 
+var DCLocations = []string{}
```
Comment (zyguan) [inline testcase/cross-region/cross-region.go:37]: If the number of DCs must be 3, then the type of `DCLocations` can be `[3]string`.

File: testcase/cross-region/tso.go (hunk 1 above)
Comment (zyguan) [inline testcase/cross-region/tso.go]: Use for loop?

File: testcase/cross-region/cross-region.go (hunk 3 above)
Comment (Thearas) [reply testcase/cross-region/cross-region.go:37]: I think the number of dc(s) may change in future, use slice here is more flexible.

File: testcase/cross-region/tso.go (hunk 2 above)
Comment (Thearas) [reply testcase/cross-region/tso.go:73]: ok

File: testcase/cross-region/cross-region.go [hunk 4]
```diff
@@ -198,7 +195,7 @@ func (c *crossRegionClient) setupDB() error {
 		return err
 	}
 	for i := 1; i <= 3; i++ {
-		stmt := fmt.Sprintf(stmtPolicy, fmt.Sprintf("p%v", i), fmt.Sprintf("dc-%v", i))
+		stmt := fmt.Sprintf(stmtPolicy, fmt.Sprintf("p%v", i), DCLocations[i-1])
```
Comment (JmPotato) [inline testcase/cross-region/cross-region.go:198]: Why not just start `i` form 0? `i-1` here seems not necessary.

Comment (Thearas) [reply testcase/cross-region/cross-region.go:198]: Because the partition name are formatting by `fmt.Sprintf("p%v", i)`.

**Review by JmPotato — APPROVED** [review]

**Review by zyguan — APPROVED** [review]
LGTM

**Comment by Thearas** [comment]
PTAL @mahjonp .

**Pull request merged**